src/backend/dx11/src/lib.rs

```diff
@@ -3748,6 +3748,36 @@ impl DescriptorSetInfo {
         }
         panic!("Unable to find binding {:?}", binding_index);
     }
+
+    fn find_uav_register(
+        &self,
+        stage: ShaderStage,
+        binding_index: pso::DescriptorBinding,
+    ) -> (DescriptorContent, RegisterData<ResourceIndex>) {
+        // Look only where uavs are stored for that stage.
+        let register_stage = if stage == ShaderStage::Compute {
+            stage
+        } else {
+            ShaderStage::Fragment
+        };
+
+        let mut res_offsets = self
+            .registers
+            .map_register(|info| info.res_index as DescriptorIndex)
+            .select(register_stage);
+        for binding in self.bindings.iter() {
```

Comment:
nit: if `self.bindings` is sorted by binding index, we could use binary search to find the match, and then check the `UAV` flag.

Reply:
That wouldn't work as we need to accumulate through all UAV bindings to get the proper offset.

Reply:
Hmm. That mapping (between bindings and resource indices) is precisely the responsibility of a pipeline layout. So I'm surprised that we have to recompute it here.